staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -697,6 +706,14 @@ func shouldSyncUpdatedNode(oldNode, newNode *v1.Node) bool {
 	if oldNode.Spec.ProviderID != newNode.Spec.ProviderID {
 		return true
 	}
+	// can happen only at the beginning of the node lifecycle
+	if oldNode.Annotations[v1.LabelTopologyRegion] != newNode.Annotations[v1.LabelTopologyRegion] {
+		return true
+	}
+	// can happen only at the beginning of the node lifecycle
+	if oldNode.Annotations[v1.LabelFailureDomainBetaZone] != newNode.Annotations[v1.LabelFailureDomainBetaZone] {
+		return true
+	}
```

Comment:
@thockin @alexanderConstantinescu you have this logic more clear than anybody, the problem is that these labels are added after the Node creation by cloud providers controllers and if these annotations are not present the implementations of loadbalancers will fail  as it happens with the Topology)

Where is the best place to add this chaeck, as a new Predicate or insdie this function?

Reply:
I would say this looks correct. The predicates express which attributes a node needs to have for it to be included in the LB set. Neither `v1.LabelTopologyRegion` or `v1.LabelTopologyZone` seem to be such required attributes (in all cases and for all cloud providers, at least). This PR seems to address the fact that we want to trigger a LB re-sync when the value of this annotation changes for any node, and for that purpose: it's implemented correctly.